### PR TITLE
server/api: show store down 

### PR DESCRIPTION
**Pull request by nolouch**
quickly reflect store down in store API.
PTAL @siddontang @disksing

### CODE REVIEW

File: server/api/store.go [hunk 1]
```diff
@@ -74,6 +74,10 @@ func newStoreInfo(store *metapb.Store, status *server.StoreStatus) *storeInfo {
 			Uptime:             typeutil.NewDuration(status.GetUptime()),
 		},
 	}
+	if time.Now().Sub(status.LastHeartbeatTS) > time.Minute {
```
Comment (siddontang) [inline server/api/store.go]: define a const var for `time.Minute`

File: server/api/store.go [hunk 2]
```diff
@@ -74,6 +74,10 @@ func newStoreInfo(store *metapb.Store, status *server.StoreStatus) *storeInfo {
...
+		s.Store.StateName = "Down"
```
Comment (siddontang) [inline server/api/store.go]: do we need to check offline or tombstone here?

Comment (nolouch) [inline server/api/store.go]: no need, those correctly reflect in the state now.

Comment (siddontang) [inline server/api/store.go]: any test to cover it? 

Or if the state is offline, can we enter this logic?

Comment (nolouch) [inline server/api/store.go]: you are right.

File: server/api/store.go [hunk 3]
```diff
@@ -74,6 +76,10 @@ func newStoreInfo(store *metapb.Store, status *server.StoreStatus) *storeInfo {
 			Uptime:             typeutil.NewDuration(status.GetUptime()),
 		},
 	}
+	if time.Now().Sub(status.LastHeartbeatTS) > defaultStoreDownTime && store.State == metapb.StoreState_Up {
+		s.Store.StateName = "Down"
```
Comment (siddontang) [inline server/api/store.go]: do we have any const var for the state name?

Comment (siddontang) [inline server/api/store.go]: how can we test it?

Comment (nolouch) [inline server/api/store.go]: maybe we need define it in `metapb`?

Comment (nolouch) [inline server/api/store.go]: maybe we need define it in `metapb`?

Comment (siddontang) [inline server/api/store.go]: I think we can construct a store and status, then call newStoreInfo and check the StateName. 

Now, I think we can define the const name in PD.

File: server/api/store.go [hunk 4]
```diff
@@ -27,6 +27,8 @@ import (
 	"github.com/unrolled/render"
 )
 
+const defaultStoreDownTime = time.Minute
```
Comment (disksing) [inline server/api/store.go]: What if the store heartbeat interval is changed to a value greater than 1 minute at tikv side?

File: server/api/store.go [hunk 5]
```diff
@@ -74,6 +76,10 @@ func newStoreInfo(store *metapb.Store, status *server.StoreStatus) *storeInfo {
 			Uptime:             typeutil.NewDuration(status.GetUptime()),
 		},
 	}
+	if time.Now().Sub(status.LastHeartbeatTS) > defaultStoreDownTime && store.State == metapb.StoreState_Up {
```
Comment (disksing) [inline server/api/store.go]: I think it will better to move this logic into `server.StoreStatus`.

**Comment by siddontang** [comment]
any update? @nolouch

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @disksing

**Comment by siddontang** [comment]
PTAL @andelf

**Comment by disksing** [comment]
LGTM. Please take a look why CI failed.

**Comment by nolouch** [comment]
report no leader, now pass.

**Pull request merged**